libarchive/archive_read_support_format_zip.c

```diff
@@ -3484,6 +3487,7 @@ read_zip64_eocd(struct archive_read *a, struct zip *zip, const char *p)
 
 	/* Save the central directory offset for later use. */
 	zip->central_directory_offset = archive_le64dec(p + 48);
+	zip->central_directory_offset_actual = zip->central_directory_offset;
```

Comment:
Should there be a TODO comment recording that this needs to be revisited?

I vaguely recall that libarchive likes to use Zip64 extensions even for small archives in some scenarios, so you should be able to test this without actually creating multi-gigabyte archives.

Reply:
Thanks for putting in that TODO comment.